packages/jira-adapter/src/filters/layouts/request_type_request_form.ts

```diff
@@ -6,10 +6,61 @@
  * CERTAIN THIRD PARTY SOFTWARE MAY BE CONTAINED IN PORTIONS OF THE SOFTWARE. See NOTICE FILE AT https://github.com/salto-io/salto/blob/main/NOTICES
  */
 
+import { Element, isInstanceElement } from '@salto-io/adapter-api'
+import { createSchemeGuard } from '@salto-io/adapter-utils'
+import Joi from 'joi'
 import { ISSUE_VIEW_TYPE, REQUEST_FORM_TYPE } from '../../constants'
 import { FilterCreator } from '../../filter'
 import { fetchRequestTypeDetails } from './layout_service_operations'
 
+type requestTypeRequestFormData = {
+  properties?: Record<string, string>
+}
+
+type requestTypeRequestFormItemComponent = {
+  data: requestTypeRequestFormData
+}
+
+const REQUEST_TYPE_REQUEST_FORM_ITEMS_COMPONENT_SCHEME = Joi.array().items(
+  Joi.object({
+    data: Joi.object({
+      properties: Joi.object().pattern(Joi.string(), Joi.string().allow('')).required(),
+    })
+      .unknown(true)
+      .required(),
+  })
+    .unknown(true)
+    .required(),
+)
+
+const isRequestTypeRequestFormItemsComponent = createSchemeGuard<requestTypeRequestFormItemComponent[]>(
+  REQUEST_TYPE_REQUEST_FORM_ITEMS_COMPONENT_SCHEME,
+  'invalid request form items component',
+)
+
+const deleteEmptyProperties = (data: requestTypeRequestFormData): void => {
+  if (data.properties) {
+    data.properties = Object.fromEntries(Object.entries(data.properties).filter(([_key, value]) => value !== ''))
+    if (Object.keys(data.properties).length === 0) {
+      delete data.properties
+    }
+  }
+}
+
+const filterEmptyProperties = (elements: Element[]): void => {
+  elements
+    .filter(e => e.elemID.typeName === REQUEST_FORM_TYPE)
+    .filter(isInstanceElement)
+    .forEach(instance => {
+      const items = instance.value.issueLayoutConfig?.items
+      if (isRequestTypeRequestFormItemsComponent(items)) {
+        items.forEach(item => {
+          deleteEmptyProperties(item.data)
+        })
+      }
+    })
+}
```

Comment:
nit: I think we can just make it a single function
```suggestion
const filterEmptyProperties = (elements: Element[]): void => {
  elements
    .filter(e => e.elemID.typeName === REQUEST_FORM_TYPE)
    .filter(isInstanceElement)
    .forEach(instance => {
      const items = instance.value.issueLayoutConfig?.items
      if (isRequestTypeRequestFormItemsComponent(items)) {
        items.forEach(({data}) => {
          if (data.properties) {
            data.properties = Object.fromEntries(Object.entries(data.properties).filter(([_key, value]) => value !== ''))
            if (Object.keys(data.properties).length === 0) {
              delete data.properties
            }
          }
        })
      }
    })
}
```

and maybe will make the name a bit more declarative, something like `removePropertiesWithEmptyValues`